### PR TITLE
[Feat] OP-wise Insight Mining

### PR DESCRIPTION
**Pull request by HYLcool**
- Support OP-wise Insight Mining to check the significance of stats/meta changes before and after each OP with T-Test measure.
- It helps users to know how each OP influences their datasets.
- Add a new measure, RelatedTTestMeasure, to measure T-Test for two related distributions on their histograms with the same bins.
- It analyzes small batches of the original dataset and the current dataset after each OP processing. After all OPs are processed, it collects all the analysis results and mines the stats/meta change significance insights from them.

>  Core related:
- Analyzer: Support analysis on str tags/categories/stats in stats and meta fields; support passing an existing Dataset object to the `run` method of analyzer instead of loading from disk.
- Monitor: Remove error messages from the resource monitor process that are caused by the main process errors, which helps to make logs clearer.
- Exporter: export meta along with stats if `export_stats` is True.

> Others
- Add two registries, Non-stats Filters and Tagging Operators, to decorate the Filters that don't produce any stats and OPs that produce tags or other infos in meta field. For now, they are:
  - Non-stats Filters: `specified_field_filter`, `specified_numeric_field_filter`, `suffix_filter`, `video_tagging_from_frames_filter` (produces tags instead of stats).
  - Tagging Operators: `video_tagging_from_frames_filter`, `image_tagging_mapper`, `video_tagging_from_audio_mapper`, `video_tagging_from_frames_mapper`

> ⚠️ **Notice**

This PR is based on the PR #512 . Please review that PR first before reviewing this one.

### CODE REVIEW

**Review by yxdyc — COMMENTED** [review]
LGTM, plz see the minor comments.

File: data_juicer/core/adapter.py [hunk 1]
```diff
@@ -177,3 +181,99 @@ def batch_size_strategy(self, load_analysis_res, base_bs=1, util_th=0.9):
...
+            curr_res = analysis_results[res_order[i]]
+
+            # only consider common stats and meta
+            common_features = list(
+                set(prev_res.features).intersection(set(curr_res.features)))
+            curr_sig_res = {}
+            for feat in common_features:
+                ttest_res = ttest_measure(prev_res[feat], curr_res[feat])
+                curr_sig_res[feat] = {
+                    't-statistic': ttest_res.statistic,
```
[the remark this answers is not in the document]
Comment (HYLcool) [reply data_juicer/core/adapter.py]: done

File: data_juicer/ops/mapper/video_tagging_from_audio_mapper.py [hunk 2]
```diff
@@ -90,5 +92,6 @@ def process_single(self, sample, rank=None):
             predicted_tag_id = torch.argmax(logits, dim=-1).item()
             predicted_tag = model.config.id2label[predicted_tag_id]
             video_audio_tags.append(predicted_tag)
-        sample[self.tag_field_name] = np.array(video_audio_tags, dtype=np.str_)
+        sample[Fields.meta][self.tag_field_name] = np.array(video_audio_tags,
```
Comment (BeachWang) [inline data_juicer/ops/mapper/video_tagging_from_audio_mapper.py:95]: Sample里面Fields.meta这个key是不是不一定存在呀？

File: data_juicer/ops/mapper/video_tagging_from_frames_mapper.py [hunk 3]
```diff
@@ -115,5 +117,5 @@ def process_single(self, sample, rank=None, context=False):
             for vid_key in videos:
                 close_video(videos[vid_key])
 
-        sample[self.tag_field_name] = video_tags
+        sample[Fields.meta][self.tag_field_name] = video_tags
```
Comment (BeachWang) [inline data_juicer/ops/mapper/video_tagging_from_frames_mapper.py:120]: 同上

File: data_juicer/ops/mapper/video_tagging_from_frames_mapper.py [hunk 4]
```diff
@@ -73,12 +74,13 @@ def __init__(self,
 
     def process_single(self, sample, rank=None, context=False):
         # check if it's generated already
-        if self.tag_field_name in sample:
+        if self.tag_field_name in sample[Fields.meta]:
```
Comment (BeachWang) [inline data_juicer/ops/mapper/video_tagging_from_frames_mapper.py:77]: Fields.meta这个key不一定存在

File: data_juicer/ops/mapper/video_tagging_from_audio_mapper.py (hunk 2 above)
Comment (HYLcool) [reply data_juicer/ops/mapper/video_tagging_from_audio_mapper.py:95]: 在这里有自动判断与添加

data_juicer/ops/base_op.py#L228-L239

**Review by yxdyc — APPROVED** [review]

**Pull request merged**